designs/enum-shapes.md

```diff
@@ -0,0 +1,259 @@
+
+```
+structure GetFooInput {
+    @httpLabel
+    suit: Suit
+}
+```
+
+### intEnum shape
+
+An intEnum is used to represent an enumerated set of integer values. The
+members of intEnum MUST be marked with the @enumValue trait set to a unique
+integer value. The following example defines an intEnum shape:
+
+```
+intEnum FaceCard {
+    @enumValue(int: 1)
+    JACK
+
+    @enumValue(int: 2)
+    QUEEN
+    
+    @enumValue(int: 3)
+    KING
+    
+    @enumValue(int: 4)
+    ACE
+    
+    @enumValue(int: 5)
+    JOKER
+}
+```
+
+#### intEnum default values
+
+The default value of the intEnum shape is 0, regardless of if the enum defines
+a member with a value of 0.
+
+```
+structure GetFooInput {
+    @default
+    suit: Suit
+}
+```
+
+intEnums MAY define a member to represent the default value of the shape by
+making the `@enumValue` of the member 0:
+
+```
+intEnum FaceCard {
+    @enumValue(0)
+    UNKNOWN
+
+    @enumValue(1)
+    JACK
+    
+    @enumValue(2)
+    QUEEN
+    
+    @enumValue(3)
+    KING
+    
+    @enumValue(4)
+    ACE
+    
+    @enumValue(5)
+    JOKER
+}
+```
+
+#### intEnum is  a specialization of integer
+
+Like enums, intEnums are considered open, meaning it is a backward compatible
+change to add new members. Previously generated clients MUST NOT fail when
+they encounter an unknown intEnum value. Client implementations SHOULD provide
+the capability of sending and receiving unknown intEnum values.
+
+To facilitate this behavior, the intEnum type is a subclass of the integer
+type. Any selector that accepts an integer implicitly accepts an intEnum.
+
+Implementation note: In Smithy’s Java implementation, shape visitors will by
+default dispatch to the integer shape when an intEnum is encountered. This
+both makes adding enums backward compatible, and allows implementations that
+do not support enums at all to ignore them.
+
+### Smithy taxonomy updates
+
+Both enum and intEnum are considered simple shapes though they have members.
+
+The current definition of aggregate shapes is:
+
+> Aggregate types define shapes that are composed of other shapes. Aggregate
+> shapes reference other shapes using members.
+
+This will require updating the description of Aggregate Types to become:
+
+> Aggregate types are shapes that can contain more than one value.
+
+The member type in Smithy is currently classified as an aggregate type. Because
+members will now be present in simple shapes, we will add a new shape type
+called “member” to go alongside simple, aggregate, and service types.
+
+## Smithy 1.0 → 2.0
+
+Since the enum trait does not require names, it is not possible to
+automatically convert string shapes using the `@enum` trait into enum traits.
+Instead, a transformer will be added that upgrades those shapes that can
+be upgraded. To make this change backward compatible for existing code
+generators, enum shapes in IDL 2.0 will always contain an enum trait with
```

Comment:
Wouldn't old generators just see a string since enum is a subtype of string? That seems forward compatible to me. It would work the same way as intEnum and integer.

Reply:
They would see a string *without the enum trait*. If somebody were to switch their string shapes to enum shapes, that would be a backwards incompatible change for generators that haven't updated.